sympy/simplify/tests/test_simplify.py

```diff
@@ -734,6 +734,16 @@ def test_issue_9324_simplify():
     assert simplify(e) == e
 
 
+def test_issue_9817_simplify():
+    from sympy.matrices.expressions import Identity, trace
+    v = MatrixSymbol('v', 3, 1)
+    A = MatrixSymbol('A', 3, 3)
+    x = Matrix([i + 1 for i in range(3)])
+    X = Identity(3)
+    quadratic = v.T * A * v
+    assert simplify((trace(quadratic.as_explicit())).xreplace({v:x, A:X})) == 14
```

Comment:
What exactly is this test for?

Reply:
As reported in the original issue, there was an `AttributeError` being thrown on trying to execute this line, i.e. trying to simplify this scalar expression 
> Finally, putting the xreplace outside works, but simplify fails on the now supposedly scalar expression.

Reply:
Can you add comments to these tests explaining what they are for?

Reply:
Sure will do, just one question. You mean adding comments that explain why these tests should pass right? Or should the comments explain what error was being thrown?

Reply:
Explain what was not working before. It doesn't look as if the simplify call is doing anything so the purpose of the test needs to be explained.